pay-api/src/pay_api/models/eft_short_names_historical.py

```diff
@@ -0,0 +1,114 @@
+    # this mapper is used so that new and old versions of the service can be run simultaneously,
+    # making rolling upgrades easier
+    # This is used by SQLAlchemy to explicitly define which fields we're interested
+    # so it doesn't freak out and say it can't map the structure if other fields are present.
+    # This could occur from a failed deploy or during an upgrade.
+    # The other option is to tell SQLAlchemy to ignore differences, but that is ambiguous
+    # and can interfere with Alembic upgrades.
+    #
+    # NOTE: please keep mapper names in alpha-order, easier to track that way
+    #       Exception, id is always first, _fields first
+    __mapper_args__ = {
+        'include_properties': [
+            'id',
+            'amount',
+            'created_by',
+            'created_on',
+            'credit_balance',
+            'description',
+            'hidden',
+            'is_processing',
+            'payment_account_id',
+            'related_group_link_id',
+            'short_name_id',
+            'statement_number',
+            'transaction_date',
+            'transaction_type'
+        ]
+    }
+    id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    amount = db.Column(db.Numeric(19, 2), nullable=False)
+    created_by = db.Column(db.String, nullable=True)
+    created_on = db.Column(db.DateTime, nullable=False, default=datetime.now(tz=timezone.utc))
+    credit_balance = db.Column(db.Numeric(19, 2), nullable=False)
+    description = db.Column(db.String, nullable=False)
+    hidden = db.Column(db.Boolean(), nullable=False, default=False, index=True)
+    is_processing = db.Column(db.Boolean(), nullable=False, default=False)
+    payment_account_id = db.Column(db.Integer, ForeignKey('payment_accounts.id'), nullable=True, index=True)
+    related_group_link_id = db.Column(db.Integer, nullable=True, index=True)
+    short_name_id = db.Column(db.Integer, ForeignKey('eft_short_names.id'), nullable=False)
+    statement_number = db.Column(db.Integer, nullable=True)
+    transaction_date = db.Column(db.DateTime, nullable=False, index=True)
+    transaction_type = db.Column(db.String, nullable=False)
+
+    @classmethod
+    def find_by_related_group_link_id(cls, group_link_id: int):
+        """Find historical records by related EFT Credit Invoice Link group id."""
+        return cls.query.filter_by(related_group_link_id=group_link_id).one_or_none()
+
+
+@define
+class EFTShortnameHistorySchema:  # pylint: disable=too-few-public-methods
+    """Main schema used to serialize an EFT Short name historical record."""
+
+    historical_id: int
+    account_id: str
+    account_name: str
+    account_branch: str
+    amount: Decimal
+    statement_number: int
+    short_name_id: int
+    short_name_balance: Decimal
+    transaction_date: datetime
+    transaction_description: str
+    transaction_type: str
+    is_processing: bool
+    is_reversible: bool
```

Comment:
When is something reversible or not?

Reply:
Reversible if:
- most recent state is that the statement has been paid and not processing
- within 60 days of that transaction date

Reply:
Will add a comment in the code

Reply:
Thanks that's what I was getting at haha